### PR TITLE
update instructions and switch imp to enum

### PR DESCRIPTION
**Pull request by MattDHill**

### CODE REVIEW

File: scripts/services/migrations.ts [hunk 1]
```diff
@@ -1,4 +1,25 @@
 import { compat, types as T } from "../deps.ts";
 
 export const migration: T.ExpectedExports.migration = compat.migrations
-    .fromMapping( { }, "0.9.7.1" );
+.fromMapping({
+    "0.9.7.1": {
+        up: compat.migrations.updateConfig(
+            (config) => {
+                const implementation = config.wallet.type === 'LndRestWallet' ? 'lnd' : 'cln'
```
Comment (islandbitcoin) [inline scripts/services/migrations.ts]: this has a bug, wallet.type is not recognized

Comment (MattDHill) [reply scripts/services/migrations.ts]: I assume you are upgrading from a prior version where the config was never set/saved?

File: docker_entrypoint.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ export LND_PATH="/mnt/lnd/admin.macaroon"
 export CLN_PATH="/mnt/c-lightning/"
 export TOR_ADDRESS=$(yq e '.tor-address' /app/data/start9/config.yaml)
 export LAN_ADDRESS=$(yq e '.lan-address' /app/data/start9/config.yaml)
-export LNBITS_BACKEND_WALLET_CLASS=$(yq e '.wallet.type' /app/data/start9/config.yaml)
+export LNBITS_BACKEND_WALLET_CLASS=$(yq e '.implementation' /app/data/start9/config.yaml)
```
Comment (islandbitcoin) [inline docker_entrypoint.sh:9]: there needs to be more changes here, the `.env` file is as follows: 

```
# Choose from LNPayWallet, OpenNodeWallet, LntxbotWallet, ClicheWallet, LnTipsWallet
#             LndRestWallet, CoreLightningWallet, LNbitsWallet, SparkWallet, FakeWallet, EclairWallet
LNBITS_BACKEND_WALLET_CLASS=LndRestWallet
```
changing the values to `lnd` and `cln` will break the `.env` file

File: scripts/services/getConfig.ts [hunk 3]
```diff
@@ -21,24 +21,15 @@ export const getConfig: T.ExpectedExports.getConfig = compat.getConfig({
...
-        "description":
-            "The LN node to connect to",
-        },
-    "default": "LndRestWallet",
-    "variants": {
-      "LndRestWallet": {},
-      "CLightningWallet":{},
-    }
+    "description": "The underlying Lightning implementation, currently LND or Core Lightning (CLN)",
+    "values": ["lnd", "cln"],
```
Comment (islandbitcoin) [inline scripts/services/getConfig.ts]:
```
# Choose from LNPayWallet, OpenNodeWallet, LntxbotWallet, ClicheWallet, LnTipsWallet
#             LndRestWallet, CoreLightningWallet, LNbitsWallet, SparkWallet, FakeWallet, EclairWallet
LNBITS_BACKEND_WALLET_CLASS=LndRestWallet
```
changing the values to `lnd` and `cln` will break the `.env` file

**Review by islandbitcoin — CHANGES_REQUESTED** [review]

File: scripts/services/migrations.ts (hunk 1 above)
Comment (islandbitcoin) [reply scripts/services/migrations.ts]: no, I cannot build the s9pk: 
```

error: TS2571 [ERROR]: Object is of type 'unknown'.
                const implementation = config.wallet.type === 'LndRestWallet' ? 'lnd' : 'cln'
                                       ~~~~~~~~~~~~~
    at file:///Users/dread/Documents/Start9/Repos/lnbits-wrapper/scripts/services/migrations.ts:8:40
make: *** [scripts/embassy.js] Error 1
```

File: docker_entrypoint.sh (hunk 2 above)
Comment (MattDHill) [reply docker_entrypoint.sh:9]: Ok, that seems like an SDK bug, but I pushed a fix that shouls satisfy it

File: scripts/services/getConfig.ts (hunk 3 above)
Comment (MattDHill) [reply scripts/services/getConfig.ts:28]: Ok, that seems like an SDK bug, but I pushed a fix that shouls satisfy it

File: docker_entrypoint.sh (hunk 2 above)
Comment (MattDHill) [reply docker_entrypoint.sh:9]: Ah, will fix

File: scripts/services/getConfig.ts (hunk 3 above)
Comment (MattDHill) [reply scripts/services/getConfig.ts]: 👍

File: docker_entrypoint.sh (hunk 2 above)
Comment (MattDHill) [reply docker_entrypoint.sh:9]: Should be fixed now

File: scripts/services/migrations.ts (hunk 1 above)
Comment (islandbitcoin) [reply scripts/services/migrations.ts]: 👍

File: docker_entrypoint.sh (hunk 2 above)
Comment (islandbitcoin) [reply docker_entrypoint.sh:9]: 👍

**Review by islandbitcoin — APPROVED** [review]
LGTM

**Pull request merged**